folly/Portability.h

```diff
@@ -205,8 +205,10 @@ FOLLY_GCC_DISABLE_NEW_SHADOW_WARNINGS
  */
 #if defined(_MSC_VER)
 #define FOLLY_TLS __declspec(thread)
-#elif defined(__GNUC__) || defined(__clang__)
+#elif (defined(__GNUC__) || defined(__clang__)) && !_LIBCPP_VER 
 #define FOLLY_TLS __thread
+#elif (defined(__GNUC__) || defined(__clang__)) && _LIBCPP_VER 
+#define FOLLY_TLS thread_local
```

Comment:
`thread_local` has different contruction and destruction semantics from both `__thread` and `__declspec(thread)`, is there some reason the version of GCC/Clang being used doesn't support `__thread`?

Reply:
This was fixed in my updated headers, pushing the change now. Thanks for the question!